documentation/api/query/v4/upgrading-from-v3.markdown

```diff
@@ -0,0 +1,125 @@
+  in that one expected use is GUI autocompletion. Then endpoint is documented
+  here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/fact-paths.html
+
+* `/v4/fact-contents` This endpoint allows fine-grained querying of facts and
+  structured facts. Documentation here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/fact-contents.html
+
+### NEW API FEATURES
+
+* The response of the reports endpoint includes the new fields `noop`,
+  `environment`, `status`, `resource_events`, `logs`, and `metrics`.  The
+  endpoint in its latest form is documented here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/reports.html and an
+  example of the new format is here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/reports.html#examples.
+  For comparison, the old response is here:
+  https://docs.puppetlabs.com/puppetdb/latest/api/query/v3/reports.html#response-format
+
+* The reports endpoint takes a latest_report? query to return only reports
+  associated with the most recent puppet run for their nodes. Similar to the
+  corresponding events query, there is no corresponding field in the response.
+  The report query fields are documented here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/reports.html#query-fields
+
+* The v4 catalogs endpoint is queryable like the other endpoints, whereas
+  before it could only return a catalog for a single host. The old query format
+  (/v4/catalogs/myhost) still works as before, but /v4/catalogs returns results
+  too. Examples are here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/catalogs.html#examples
+
+* Extract is available as a top-level query operator, useful for selecting only
+  certain fields from a response. Documentation is here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/operators.html#extract
+
+* The v4 events endpoint does not require a query parameter, so /v4/events is
+  now a valid query. The endpoint is documented here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/events.html#get-v4events
+
+* The new select_fact_contents subquery operator allows for filtering the
+  results of other endpoints based on detailed queries about structured fact
+  values. This is exhibited on the bottom of the page here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/operators.html#subquery-examples
+
+* We have added the regexp array match operator `~>` for querying fact paths on
+  the `fact-contents` or `fact-paths endpoints`. This is documented here:
+  https://docs.puppetlabs.com/puppetdb/master/api/query/v4/operators.html#regexp-array-match.
+  An example of usage is given at the bottom of the subquery-examples page
+  linked just above.
+
+* The `in` and `extract` operators have been changed to accept multiple fields,
+  allowing more concise subquerying as explained here:
+  https://github.com/puppetlabs/puppetdb/pull/1053
+
+### CHANGES STILL TO COME
```

Comment:
Seems like we can leave this out. For one we don't want to guarantee, also some of this might not even come in 3.0.0. Others like the state-overview endpoint won't be in this repo.